TeamCode/src/main/java/org/firstinspires/ftc/teamcode/freightfrenzy2021/opmodes/AutonOpModeV2.java

```diff
@@ -140,4 +152,39 @@ private void updateTelemetry(){
         telemetry.update();
     }
 
+    //Open up the camera for Freight Detection
+    private void startCamera(){
+        freightDetector = new FreightDetector();
+
+        //No Longer needed when using overloaded function createWebcam(webCabName)
+        //int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        //Log.d(logTag, "cameraMonitorViewId set");
+        EbotsWebcam bucketWebCam = new EbotsWebcam(hardwareMap, "bucketCam", RobotSide.FRONT, 0,-3.25f, 9.0f);
+        WebcamName webcamName = bucketWebCam.getWebcamName();
+        // With live preview
+        //why did we change from createWebcam(webCabName, viewportContainerId) as to createWebcam(webCabName)
+        camera = OpenCvCameraFactory.getInstance().createWebcam(webcamName);
+        Log.d(logTag, "camera instantiated");
+        camera.openCameraDeviceAsync(new OpenCvCamera.AsyncCameraOpenListener()
+        {
+            @Override
+            public void onOpened()
+            {
+                Log.d(logTag, "The camera is now open...");
+                camera.startStreaming(320, 240, OpenCvCameraRotation.UPRIGHT);
+                camera.setPipeline(freightDetector);
+
+            }
+            @Override
+            public void onError(int errorCode)
+            {
+                Log.d(logTag, "There was an error");
+            }
+        });
+        Log.d(logTag, "Camera for Freight Detector Instantiated");
+
+        Log.d(logTag, "Constructor complete");
+
+    }
+
 }
```

Comment:
You need to add a getter function for the FreightDetector object which can be accessed by the AutonState